Update LSP23 guide to use @erc725.js (#903)
lukso-network/docs: docs/learn/expert-guides/deploy-up-with-lsp23.md

Add to your file the imports of the artifacts, constants and libraries that we will use:

```typescript
// libs
import { AbiCoder, Contract, ethers } from 'ethers';
import { ERC725 } from '@erc725/erc725.js';

> // LSPs artifacts

Reviewer: ```suggestion
// LSPs Smart Contracts artifacts
```

#### Create the LSP3 Metadata

The LSP3 Metadata is a JSON object that contains information about the Universal Profile. For more information on the LSP3 Metadata, please refer to the [LSP3 Profile Metadata specification](../../standards/universal-profile/lsp3-profile-metadata.md). For the sake of simplicity, we will use one that we previously created and pushed to IPFS:

```typescript
async function main() {
  // previous code

  // instantiate the erc725 class
  const erc725 = new ERC725([
    ...LSP6KeyManagerSchemas,
    ...LSP3ProfileMetadataSchemas,
    ...LSP1UniversalReceiverDelegateSchemas,
  ]);

  const lsp3DataValue = {
    verification: {
      method: 'keccak256(utf8)',
      data: '0x6d6d08aafb0ee059e3e4b6b3528a5be37308a5d4f4d19657d26dd8a5ae799de0',
    },

>     url: 'ipfs://QmPRoJsaYcNqQiUrQxE7ajTRaXwHyAU29tHqYNctBmK64w',

Reviewer: Maybe would be good to mention in comment here to replace with the actual user values?

  };
}
```

#### Create the permissions data keys and data values

> In order to create all the permissions data keys and data values, we will use the `@erc725/erc725.js` library. This library is a JavaScript implementation of the ERC725Y standard. It allows you to easily encode and decode the data keys and data values of an ERC725Y contract. For more information on this library, please refer to the [ERC725Y.js](../../tools/erc725js/getting-started.md) docs.

Reviewer: Why ERC725Y.js?

Suggest following change to shorten sentence.

```suggestion
In order to create all the permissions data keys and data values, we will use the [`@erc725/erc725.js`](../../tools/erc725js/getting-started.md) library. This library is a JavaScript implementation to encode and decode data key and values easily from any ERC725Y contract storage.
```

```typescript
async function main() {
  // previous code

  // create the permissions data keys
  const setDataKeysAndValues = erc725.encodeData([
    { keyName: 'LSP3Profile', value: lsp3DataValue }, // LSP3Metadata data key and value
    {
      keyName: 'LSP1UniversalReceiverDelegate',
      value: UNIVERSAL_RECEIVER_ADDRESS,
    }, // Universal Receiver data key and value
    {
      keyName: 'AddressPermissions:Permissions:<address>',
      dynamicKeyParts: [UNIVERSAL_RECEIVER_ADDRESS],
      value: erc725.encodePermissions({
        REENTRANCY: true,
        SUPER_SETDATA: true,
      }),
    }, // Universal Receiver Delegate permissions data key and value
    {
      keyName: 'AddressPermissions:Permissions:<address>',
      dynamicKeyParts: [MAIN_CONTROLLER],

>       value: erc725.encodePermissions({

Reviewer: Maybe for the main controller permissions we could re-use the constant from the lsp-smart-contracts? But keeping it like this is also very good IMO and very clear. So I would even argue and leave it as it is.

Maybe for the main controller permissions we could re-use the constant from the lsp-smart-contracts? But keeping it like this is also very good

Author: hmm i am not sure i like the fact that they can choose which permisison they want to give at deployment

Reviewer: OK we agreed to keep it like this as it makes it crystal clear 👌

        CHANGEOWNER: true,
        ADDCONTROLLER: true,
        EDITPERMISSIONS: true,
        ADDEXTENSIONS: true,
        CHANGEEXTENSIONS: true,
        ADDUNIVERSALRECEIVERDELEGATE: true,
        CHANGEUNIVERSALRECEIVERDELEGATE: true,
        REENTRANCY: false,
        SUPER_TRANSFERVALUE: true,
        TRANSFERVALUE: true,
        SUPER_CALL: true,
        CALL: true,
        SUPER_STATICCALL: true,
        STATICCALL: true,
        SUPER_DELEGATECALL: false,
        DELEGATECALL: false,
        DEPLOY: true,
        SUPER_SETDATA: true,
        SETDATA: true,
        ENCRYPT: true,
        DECRYPT: true,
        SIGN: true,
        EXECUTE_RELAY_CALL: true,
      }), // Main Controller permissions data key and value
    },

>     // length of the Address Permissions array and their respective indexed keys and values

Reviewer: ```suggestion
   // Address Permissions array length = 2, and the controller addresses at each index
```

#### Create the LSP3 Metadata

The LSP3 Metadata is a JSON object that contains information about the Universal Profile. For more information on the LSP3 Metadata, please refer to the [LSP3 Profile Metadata specification](../../standards/universal-profile/lsp3-profile-metadata.md). For the sake of simplicity, we will use one that we previously created and pushed to IPFS:

```typescript
async function main() {
  // previous code

  // instantiate the erc725 class
  const erc725 = new ERC725([
    ...LSP6KeyManagerSchemas,
    ...LSP3ProfileMetadataSchemas,
    ...LSP1UniversalReceiverDelegateSchemas,
  ]);

  const lsp3DataValue = {
    verification: {
      method: 'keccak256(utf8)',
      data: '0x6d6d08aafb0ee059e3e4b6b3528a5be37308a5d4f4d19657d26dd8a5ae799de0',
    },

>     // this is an IPFS CID of a LSP3 Metadata example, you can use your own

Reviewer: ```suggestion
    // this is an IPFS CID of a LSP3 Profile Metadata example, you can use your own
```

async function main() {
  // previous code

  const abiCoder = new AbiCoder();

>   const types = ['bytes32[]', 'bytes[]']; // types of the parameters

Reviewer: Is this variable re-used? I would inline with next line for simplicity
